src/Microsoft.AspNetCore.Razor.Design/Microsoft.AspNetCore.Razor.Design.csproj

```diff
@@ -2,8 +2,68 @@
 
   <PropertyGroup>
     <Description>Razor is a markup syntax for adding server-side logic to web pages. This package contains MSBuild support for Razor.</Description>
-    <TargetFrameworks>net46;netstandard2.0</TargetFrameworks>
+
+    <!-- 
+      Using netcoreapp2.0 here is a workaround for https://github.com/Microsoft/msbuild/issues/2661
+
+      We need to use a P2P reference to build some other projects that need to target netcoreapp2.0
+      and even though we don't actually want to reference the output, MSBuild will not allow it.
+    -->
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+
+    <!-- This project doesn't have any code, so don't include it in the .nupkg -->
+    <IncludeBuildOutput>false</IncludeBuildOutput>
   </PropertyGroup>
 
+  <!-- 
+    We're using CopyToOutputDirectory and Link here to mimic shape of what goes into the package in the output
+    directory. This is used by tests.
+  -->
+  <ItemGroup>
+    <Content Include="build\**\*.props" PackagePath="build\" />
+    <Content Include="build\**\*.targets" PackagePath="build\" />
+    <Content Include="buildMultiTargeting\*.props" PackagePath="buildMultiTargeting\" />
+    <Content Include="buildMultiTargeting\*.targets" PackagePath="buildMultiTargeting\" />
+
+    <Content Include="..\Microsoft.AspNetCore.Razor.Tasks\bin\$(Configuration)\net46\Microsoft.AspNetCore.Razor.Tasks.dll">
+      <PackagePath>tasks\net46\</PackagePath>
+      <Link>tasks\net461\Microsoft.AspNetCore.Razor.Tasks.dll</Link>
+      <Hidden>true</Hidden>
+    </Content>
+    <Content Include="..\Microsoft.AspNetCore.Razor.Tasks\bin\$(Configuration)\netstandard2.0\Microsoft.AspNetCore.Razor.Tasks.dll">
+      <PackagePath>tasks\netstandard2.0\</PackagePath>
+      <Link>tasks\netstandard2.0\Microsoft.AspNetCore.Razor.Tasks.dll</Link>
+      <Hidden>true</Hidden>
+    </Content>
+
+    <Content Include="..\Microsoft.AspNetCore.Razor.GenerateTool\bin\$(Configuration)\netcoreapp2.0\publish\**\*.*">
+      <PackagePath>tools\%(RecursiveDir)\%(Filename)%(Extension)</PackagePath>
+      <Link>tools\%(RecursiveDir)\%(Filename)%(Extension)</Link>
+      <Hidden>true</Hidden>
+    </Content>
```

Comment:
This is also having an issue on clean builds, I'll dig in here.

Reply:
Most likely because the files don't exist when the glob is evaluated. You have to delay the glob evaluation after /t:BuildProjectReferences are built and before /t:GenerateNuspec

Reply:
Or, use nuspec.

Reply:
I don't want to write the `.nuspec` because it will contain files we don't own.